File: solutionFiles/LongestCommonPrefix.py
import json
import sys

def longest_common_prefix(strs):
    if not strs:
        return ""

    prefix = strs[0]
    for string in strs[1:]:
        while string[:len(prefix)] != prefix:
            prefix = prefix[:-1]
            if not prefix:
                return ""
    return prefix

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print(json.dumps({"error": "No input provided"}))
        sys.exit(1)

    input_str = sys.argv[1]
    try:
        strs = json.loads(input_str)
        if not isinstance(strs, list) or not all(isinstance(s, str) for s in strs):
            raise ValueError
    except ValueError:
        print(json.dumps({"error": "Invalid input"}))
        sys.exit(1)

    result = longest_common_prefix(strs)
    print(json.dumps({"result": result}))

File: solutionFiles/LongestCommonPrefix.js
const longestCommonPrefix = (strs) => {
    if (!strs.length) return "";

    let prefix = strs[0];
    for (let i = 1; i < strs.length; i++) {
        while (strs[i].indexOf(prefix) !== 0) {
            prefix = prefix.slice(0, -1);
            if (!prefix) return "";
        }
    }
    return prefix;
};

const main = () => {
    if (process.argv.length < 3) {
        console.error(JSON.stringify({ error: "No input provided" }));
        process.exit(1);
    }

    const inputStr = process.argv[2];
    let strs;
    try {
        const data = JSON.parse(inputStr);
        if (!data || typeof data !== 'object' || !Array.isArray(data.strs)) {
            throw new Error();
        }
        strs = data.strs;
        if (!strs.every(s => typeof s === 'string')) {
            throw new Error();
        }
    } catch (e) {
        console.error(JSON.stringify({ error: "Invalid input" }));
        process.exit(1);
    }

    const result = longestCommonPrefix(strs);
    console.log(JSON.stringify({ longest_common_prefix: result }));
};

main();
